lib/fizzy/execute.cpp

```diff
@@ -388,6 +388,27 @@ execution_result execute(Instance& instance, FuncIdx func_idx, std::vector<uint6
                 stack.push(ret.stack[0]);
             break;
         }
+        case Instr::return_:
+        {
+            // TODO: Not needed, but satisfies the assert in the end of the main loop.
+            labels.clear();
+
+            assert(func_idx < instance.module.funcsec.size());
+            const auto type_idx = instance.module.funcsec[func_idx];
+            assert(type_idx < instance.module.typesec.size());
```

Comment:
I think actually these two could also be moved up if they are used multiple times?

Reply:
This is used at most once.